### ISSUE TITLE
why need sparse block diagonal adjacency matrices at all

### ISSUE THREAD

**Issue by ajayaraman** [tkipf/gcn#105]
I'm wondering why sparse block diagonal matrices are needed in the first place. 

Would it not be simpler to have a mini-batch of adjacency matrices for a graph with nV vertices
A = [N x nV x nV]
W = [nHidden x nV]

then we can simply do
H = tf.nn.relu(   tf.reshape( tf.dot( tf.reshape(A, -1, nV) , W), -1, nV, nV)  )

**Comment by tkipf** [comment]
Yes, but this assumes that A is a dense matrix, hence the computational
complexity is much higher.

**Comment by ajayaraman** [comment]
I see. So the sparse adjacency matrix is more to support large graphs with sparse edge connectivity. Not so much mini-batching.

**Comment by ajayaraman** [comment]
I see. So the sparse adjacency matrix is more to support large graphs with sparse edge connectivity. Not so much mini-batching.

**Comment by ajayaraman** [comment]
I see. So the sparse adjacency matrix is more to support large graphs with sparse edge connectivity. Not so much mini-batching.
